src/urllib3/http2.py

```diff
@@ -72,11 +72,19 @@ def putrequest(
                 )
             )
 
-    def putheader(self, header: str, *values: str) -> None:
+    def putheader(self, header: str, *values: str) -> None:  # type: ignore[override]
         for value in values:
-            self._h2_headers.append(
-                (header.encode("utf-8").lower(), value.encode("utf-8"))
-            )
+            if isinstance(header, str):
+                header_bytes = header.encode("utf-8").lower()
+            else:
+                header_bytes = header.lower()
+
+            if isinstance(value, str):
+                value_bytes = value.encode("utf-8")
+            else:
+                value_bytes = value
+
+            self._h2_headers.append((header_bytes, value_bytes))
```

Comment:
Let's move adding support for non-str values out this PR

Reply:
shall we throw error for other types? or add ignore @illia-v

Reply:
In this PR no functional changes (new errors, etc.) are wanted because we don't add support for any new types

Reply:
Cool will remove the